tests/vlan/test_vlan.py

```diff
@@ -35,71 +33,141 @@ def cfg_facts(duthosts, rand_one_dut_hostname):
+            if vlan_cnt >= 2:
+                break
+    return vlan_intfs_dict
 
 
 @pytest.fixture(scope="module")
-def vlan_ports_list(rand_selected_dut, tbinfo, cfg_facts):
+def vlan_ports_list(duthosts, rand_one_dut_hostname, rand_selected_dut, tbinfo, cfg_facts, vlan_intfs_dict):
+    duthost = duthosts[rand_one_dut_hostname]
     mg_facts = rand_selected_dut.get_extended_minigraph_facts(tbinfo)
     vlan_ports_list = []
     config_ports = {k: v for k,v in cfg_facts['PORT'].items() if v.get('admin_status', 'down') == 'up'}
     config_portchannels = cfg_facts.get('PORTCHANNEL', {})
     config_port_indices = {k: v for k, v in mg_facts['minigraph_ptf_indices'].items() if k in config_ports}
     ptf_ports_available_in_topo = {port_index: 'eth{}'.format(port_index) for port_index in config_port_indices.values()}
-    config_port_channel_members = [port_channel[1]['members'] for port_channel in config_portchannels.items()]
+    config_port_channel_members = [port_channel['members'] for port_channel in config_portchannels.values()]
     config_port_channel_member_ports = list(itertools.chain.from_iterable(config_port_channel_members))
+    config_ports_vlan = defaultdict(list)
+    vlan_members = cfg_facts.get('VLAN_MEMBER', {})
+    for k, v in cfg_facts['VLAN'].items():
+        vlanid = v['vlanid']
+        for addr in cfg_facts['VLAN_INTERFACE']['Vlan'+vlanid]:
+            if addr.find(':') == -1:
+                ip = addr
+                break
+        else:
+            continue
+        for port in v['members']:
+            if k in vlan_members and port in vlan_members[k]:
+                mode = vlan_members[k][port]['tagging_mode']
+                config_ports_vlan[port].append((int(vlanid), ip, mode))
+
+    vlan_id_list = [k for k, v in vlan_intfs_dict.items() if v['orig'] == False]
     pvid_cycle = itertools.cycle(vlan_id_list)
     # when running on t0 we can use the portchannel members
     if config_portchannels:
-        for po in config_portchannels.keys()[:2]:
+        portchannel_cnt = 0
+        for po in config_portchannels:
             port = config_portchannels[po]['members'][0]
-            vlan_ports_list.append({
+            vlan_port = {
                 'dev' : po,
                 'port_index' : [config_port_indices[member] for member in config_portchannels[po]['members']],
-                'pvid' : pvid_cycle.next(),
-                'permit_vlanid' : { vid : {
-                    'peer_ip' : '192.168.{}.{}'.format(vid, 2 + config_port_indices.keys().index(port)),
-                    'remote_ip' : '{}.1.1.{}'.format(vid, 2 + config_port_indices.keys().index(port))
-                    } for vid in vlan_id_list }
-            })
+                'permit_vlanid' : {}
+            }
+            if tbinfo['topo']['name'] == 't0-56-po2vlan' and po in config_ports_vlan:
+                vlan_port['pvid'] = 0
+                for vlan in config_ports_vlan[po]:
+                    if vlan[2] == 'untagged':
```

Comment:
>[2] [](http://example.com/codeflow?start=27&length=3)

Can you remove magic number. How to make sure index 2 is available?

Reply:
vlan is a tuple, ((int(vlanid), ip, mode)).
Will use dict to remove magic number.